test/typescript/providers/test.ts

```diff
@@ -9,6 +9,114 @@ describe("full integration test", () => {
   });
 
   test("build providers", () => {
-    expect(driver.deploy()).toMatchSnapshot();
+    expect(driver.deploy("using-all-providers")).toMatchSnapshot();
+  });
+
+  describe("namespaces", () => {
+    let stack;
+    beforeAll(() => {
+      stack = driver.synthesizedStack("namespaces");
+    });
+
+    test("has resources", () => {
+      expect(stack.byId("instance")).toBeTruthy();
+      expect(stack.byId("lambdaFn")).toBeTruthy();
+    });
+
+    test("has data source", () => {
+      expect(stack.byId("ami")).toBeTruthy();
+      expect(stack.byId("callerIdentity")).toBeTruthy();
+    });
+  });
+
+  describe("provider functionality", () => {
+    beforeAll(async () => {
+      await driver.synth();
+    });
+
+    describe("references", () => {
+      let stack;
+      beforeAll(() => {
+        stack = driver.synthesizedStack("references");
+      });
+
+      // HCL does not support referencing provider values
+      test("provider references are no references", () => {
+        const providerAddress = stack.output("provideraddress");
+        expect(providerAddress).toEqual("http://127.0.0.1");
+      });
+
+      test("simple references", () => {
+        const jobSpec = stack.byId("secondJob").jobspec;
+        expect(jobSpec).toContain("${");
+        expect(jobSpec).toContain("firstJob");
+        expect(jobSpec).toContain("jobspec");
+      });
+
+      test("single-item references", () => {
+        const namespace = stack.byId("myDeployment").metadata.namespace;
+        expect(namespace).toContain("${");
+        expect(namespace).toContain("myNamespace");
+        expect(namespace).toContain(".metadata[0].name");
+      });
+    });
+
+    describe("mutation", () => {
+      describe("Simple", () => {
+        let instance;
+        beforeAll(() => {
+          instance = driver.synthesizedStack("mutation").byId("instance");
+        });
+
+        test("direct primitive mutation", () => {
+          expect(instance.instance_type).toBe("t2.small");
+        });
+
+        test("direct reference mutation", () => {
+          expect(instance.ami).toEqual("${data.aws_ami.ami.id}");
+        });
+
+        test("reset method mutation", () => {
+          expect(instance.availability_zone).not.toBeDefined();
+        });
+
+        test("put method mutation", () => {
+          expect(instance.metadata_options).toEqual({
+            http_endpoint: "127.0.0.1",
+          });
+        });
+      });
+
+      // The necessary functions are exposed, but currently do not work
```

Comment:
maybe link to an issue? I wouldn't know why they don't work or where to look whether this comment is still valid.